hazelcast/src/main/java/com/hazelcast/instance/HazelcastProperties.java

```diff
@@ -55,34 +56,46 @@
      * @param hazelcastProperties array of {@link HazelcastProperty} to configure
      */
     protected void initProperties(Properties properties, HazelcastProperty[] hazelcastProperties) {
+        this.properties = properties;
         for (HazelcastProperty property : hazelcastProperties) {
```

Comment:
A user should not be.

This is more a question about responsible programming. The question is if we need to protect the exposing collections and perhaps I should protect it. 

But @tombujok if you are fine with it; I can do this cleanup while dealing with the HazelcastProperties. I can get rid of the Properties completley and expose a keySet method which returns a immutable set and a getValue method. This way nobody can by accident modify a Properties-instance while accessing the HazelcastProperties